### PR TITLE
[Merged by Bors] - chore(SetTheory/Cardinal/Basic): organize file

### PR DESCRIPTION
**Pull request by vihdzp**
This PR moves related theorems together, and adds captions for each section. Superfluous `section`, `open`, `universe` and `variable` are deleted.

We also move the proof of the well-ordering theorem from the `Ordinal/Basic` file to here, where it makes more sense and can be potentially used to prove other basic results. This has the nice effect of removing a bunch of imports to the file.

No theorem statements were modified, aside from the very minor change of correcting `∀` to `Π` in `prod` and `mk_pi`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by vihdzp** [comment]
@YaelDillies This would be nice to have before doing the `Cardinal.IsLimit` refactor, it makes it a bit easier to keep track of everything in that file.

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 1]
```diff
@@ -733,6 +818,9 @@ theorem isSuccPrelimit_zero : IsSuccPrelimit (0 : Cardinal) :=
 @[deprecated isSuccPrelimit_zero (since := "2024-09-05")]
 alias isSuccLimit_zero := isSuccPrelimit_zero
 
+/-! ### Indexed cardinal `sum` -/
+
+
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:823]:
```suggestion
```

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 2]
```diff
@@ -804,6 +895,44 @@ theorem lift_mk_le_lift_mk_mul_of_lift_mk_preimage_le {α : Type u} {β : Type v
...
+theorem sum_nat_eq_add_sum_succ (f : ℕ → Cardinal.{u}) :
+    Cardinal.sum f = f 0 + Cardinal.sum fun i => f (i + 1) := by
+  refine (Equiv.sigmaNatSucc fun i => Quotient.out (f i)).cardinal_eq.trans ?_
+  simp only [mk_sum, mk_out, lift_id, mk_sigma]
+
+end Cardinal
+
+/-! ### Well-ordering theorem -/
+
+
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:907]:
```suggestion
```

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 3]
```diff
@@ -804,6 +895,44 @@ theorem lift_mk_le_lift_mk_mul_of_lift_mk_preimage_le {α : Type u} {β : Type v
...
+instance WellOrderingRel.isWellOrder : IsWellOrder α WellOrderingRel :=
+  (RelEmbedding.preimage _ _).isWellOrder
+
+instance IsWellOrder.subtype_nonempty : Nonempty { r // IsWellOrder α r } :=
+  ⟨⟨WellOrderingRel, inferInstance⟩⟩
+
+/-! ### Small sets of cardinals -/
+
+
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:933]:
```suggestion
```

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 4]
```diff
@@ -846,8 +975,8 @@ theorem bddAbove_range_comp {ι : Type u} {f : ι → Cardinal.{v}} (hf : BddAbo
   rw [range_comp]
   exact bddAbove_image g hf
 
-theorem iSup_le_sum {ι} (f : ι → Cardinal) : iSup f ≤ sum f :=
-  ciSup_le' <| le_sum _
+/-! ### Bounds on suprema -/
+
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:979]:
```suggestion
```

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 5]
```diff
@@ -892,29 +1059,16 @@ lemma exists_eq_of_iSup_eq_of_not_isLimit
   rw [← le_zero_iff] at h ⊢
   exact (le_ciSup hf _).trans h
 
--- Porting note: simpNF is not happy with universe levels.
-@[simp, nolint simpNF]
-theorem lift_mk_shrink (α : Type u) [Small.{v} α] :
-    Cardinal.lift.{max u w} #(Shrink.{v} α) = Cardinal.lift.{max v w} #α :=
-  lift_mk_eq.2 ⟨(equivShrink α).symm⟩
+/-! ### Indexed cardinal `prod` -/
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:1063]:
```suggestion
```
etc

File: Mathlib/SetTheory/Cardinal/Basic.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Basic.lean:823]: I thought two spaces after each section was the new convention...

Comment (YaelDillies) [reply Mathlib/SetTheory/Cardinal/Basic.lean]: That's just what mathport did to our files but I don't think anyone is following that convention

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 6]
```diff
@@ -83,10 +83,13 @@ open Function Set Order
 
 noncomputable section
 
-universe u v w
+universe u v w v' w'
 
 variable {α β : Type u}
 
+/-! ### Definition of cardinals -/
+
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Basic.lean:91]:
```suggestion
```
and below

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**